genie-agent/src/main/java/com/netflix/genie/agent/utils/EnvUtils.java

```diff
@@ -0,0 +1,173 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.utils;
+
+import com.google.common.collect.Maps;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Utilities for shell environment variables.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+public final class EnvUtils {
+
+    private static Pattern singleLineValuePattern = Pattern.compile("^(.+)='(.*)'$");
+    private static Pattern multiLineValueBeginPattern = Pattern.compile("^(.+)='(.*)$");
+    private static Pattern multiLineValueEndPattern = Pattern.compile("^(.*)[']$");
+    private static Pattern multiLineValueMiddlePattern = Pattern.compile("^([^']*)$");
+
+    private EnvUtils() {
+    }
+
+    /**
+     * Parse a file containing shell variable declarations.
+     * Expects all values to be single-quoted. Lines not conforming are silently skipped.
+     * Supports multi-line values.
+     *
+     * @param environmentFile file to parse
+     * @return a map of the values parsed
+     * @throws IOException if the file cannot be read
+     * @throws ParseException if the file contents cannot be parsed
+     */
+    public static Map<String, String> parseEnvFile(
+        final File environmentFile
+    ) throws IOException, ParseException {
+        return parseEnvStream(new FileInputStream(environmentFile));
+    }
+
+    /**
+     * Parse a file containing shell variable declarations.
+     * Expects all values to be single-quoted. Lines not conforming are silently skipped.
```

Comment:
I have not, i should crawl the genie api for `env.txt`.
Even if we don't find any tho, it's inconclusive because all the runner jobs cannot be looked at this way.

I'd go ahead without this for the moment, we can make a note to add "escaped single quote support" to the env parser later.